lib/archethic/self_repair/notifier.ex

```diff
@@ -105,7 +118,88 @@ defmodule Archethic.SelfRepair.Notifier do
     {:noreply, state}
   end
 
-  defp repair_transactions(node_key, current_node_public_key) do
+  @spec repair_beacon_summary_aggregates(
+          absent_node_public_key :: Crypto.key(),
+          current_node_public_key :: Crypto.key()
+        ) :: :ok
+  def repair_beacon_summary_aggregates(absent_node_public_key, current_node_public_key) do
+    Logger.debug("Trying to repair shard unavailablity due to a topology change",
+      node: Base.encode16(absent_node_public_key)
+    )
+
+    Utils.genesis_node_enrollment_date()
+    # stream of next summary dates
+    |> SummaryTimer.next_summaries()
+    # get aggregate address and prev chain Storage nodes
+    |> Stream.map(&summary_aggregates_to_sync(&1))
+    # keep summary_aggreagate_addr if absent_node_public_key is present in prev chain_storage_nodes
+    |> Stream.filter(&Utils.key_in_node_list?(elem(&1, 1), absent_node_public_key))
+    |> Stream.filter(&(&1 |> elem(1) != []))
+    |> Stream.map(&sync_unavailable_summary_aggregate(&1, current_node_public_key))
+    |> Stream.run()
+  end
+
+  @spec summary_aggregates_to_sync(summary_date :: DateTime.to()) ::
+          {summary_date :: DateTime.t(), previous_chain_storage_nodes :: list(Node.t())}
+  defp summary_aggregates_to_sync(summary_date) do
+    node_list =
+      P2P.list_nodes()
+      |> Enum.filter(fn node ->
+        node.authorization_date != nil &&
+          DateTime.compare(node.authorization_date, summary_date) == :lt
+      end)
+
+    previous_chain_storage_nodes =
+      summary_date
+      |> Crypto.derive_beacon_aggregate_address()
+      |> Election.chain_storage_nodes(node_list)
+
+    {summary_date, previous_chain_storage_nodes}
+  end
+
+  @spec sync_unavailable_summary_aggregate(
+          {summary_date :: DateTime.t(), previous_chain_storage_nodes :: list(Node.t())},
+          current_node_public_key :: Crypto.key()
+        ) :: :ok
+  def sync_unavailable_summary_aggregate(
+        {summary_date, previous_chain_storage_nodes},
+        current_node_public_key
+      ) do
+    new_chain_storage_nodes =
+      Election.chain_storage_nodes(
+        Crypto.derive_beacon_aggregate_address(summary_date),
+        P2P.authorized_nodes() -- previous_chain_storage_nodes
+      )
```

Comment:
I think you should subtract  the previous chain storage nodes after, and it should return either empty or single node list.
```suggestion
      new_chain_storage_nodes =
      Election.chain_storage_nodes(
        Crypto.derive_beacon_aggregate_address(summary_date),
        P2P.authorized_nodes() |> Enum.reject(& &1.first_public_key == absent_node_public_key)
      ) -- previous_chain_storage_nodes
```